### PR TITLE
[mlir][tensor] Fix `tensor.pad` to remove newly static values

### PR DESCRIPTION
**Pull request by rsuderman**
The canonicalization incrementally converts foldable dynamic hi/lo padding to static hi/lo values. During this canonicalization the static-fied valued should be removed from the dynamic values.

### CODE REVIEW

**Review by hanhanW — APPROVED** [review]

File: mlir/lib/Dialect/Tensor/IR/TensorOps.cpp [hunk 1]
```diff
@@ -3158,19 +3158,23 @@ struct FoldStaticPadding : public OpRewritePattern<PadOp> {
 
     // Extract the static info from the high and low operands.
     SmallVector<int64_t> constOperandsLow;
+    llvm::SmallVector<Value> newLows;
```
Comment (kuhar) [inline mlir/lib/Dialect/Tensor/IR/TensorOps.cpp:3161]: nit: You don't need the `llvm::`, there's a using declaration for this and some other common types in the mlir namespace. Also below.

File: mlir/test/Dialect/Tensor/canonicalize.mlir [hunk 2]
```diff
@@ -1361,7 +1361,7 @@ func.func @pad_same_static_shape(%arg0: tensor<5x6xf32>, %a: index)
 // CHECK-LABEL:   func @pad_fold_static(
 // CHECK-SAME:      %[[INPUT:.*]]: tensor<?x64x?x?xf32>) -> tensor<?x?x?x?xf32> {
 // CHECK:           %[[CST:.*]] = arith.constant 0.000000e+00 : f32
-// CHECK:           %[[PADDING:.*]] = arith.constant 4 : index
+// CHECK-NOT:       %[[PADDING:.*]] = arith.constant 4 : index
```
Comment (kuhar) [inline mlir/test/Dialect/Tensor/canonicalize.mlir:1364]: nit: since this is checking for absence, do we want to capture the name?

**Review by kuhar — APPROVED** [review]
LGTM % nits

File: mlir/lib/Dialect/Tensor/IR/TensorOps.cpp (hunk 1 above)
Comment (rsuderman) [reply mlir/lib/Dialect/Tensor/IR/TensorOps.cpp]: Done.

File: mlir/test/Dialect/Tensor/canonicalize.mlir (hunk 2 above)
Comment (rsuderman) [reply mlir/test/Dialect/Tensor/canonicalize.mlir]: Done.

**Pull request merged**